lib/internal/http2/compat.js

```diff
@@ -497,4 +522,37 @@ function onServerStream(stream, headers, flags) {
   server.emit('request', request, response);
 }
 
+class Http2PushRequest {
+  constructor(stream, headers) {
+    this[kStream] = stream;
+    this[kHeaders] = headers || Object.create(null);
+    this[kState] = {
+      headersSent: false
+    };
+  }
+
+  push(callback) {
+    var state = this[kState];
+    if (state.headersSent === true)
+      throw new Error('Cannot push request headers ' +
+        'after the HTTP message has been sent');
+    var stream = this[kStream];
+    if (stream === undefined)
+      throw new Error('HTTP/2 Stream has been closed');
+    var headers = this[kHeaders];
+    try {
+      stream.pushStream(
+        headers,
+        function(promisedStream, headers, options) {
+          var response = new Http2ServerResponse(promisedStream);
+          callback(null, response);
+        }
+      );
+      state.headersSent = true;
+    } catch (error) {
+      callback(error);
+    }
+  }
+}
```

Comment:
Agreed. I don't know what I was thinking. 🤦‍♂️

c6a2fe2 implements the API as requested. Also renamed to `createPushResponse` since the returned object is an `Http2ServerResponse`.